### ISSUE TITLE
Unclosed client session

### ISSUE THREAD

**Issue by prashant-nigam** [Azure/azure-kusto-python#417]
#### Code Sample

```python
async with KustoClient(kcsb) as client:
    response = await client.execute(dbname, query,properties=clientReqProperties)


```
#### Problem description

I am initializing client using 'with' but still getting below errors:

[Error] Unclosed client sessionclient_session: <aiohttp.client.ClientSession object at 0x7ffa30df8310>
2022-09-07T10:10:42.288 
[Error] Unclosed connectorconnections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7ffa30e136e0>, 946021.36840352)]']connector: <aiohttp.connector.TCPConnector object at 0x7ffa30df84d0>

I tried adding below code but still getting error:
```python
await client._session.close()
```

#### Output of ``pip freeze`` 

<details>

azure-functions==1.11.2
pyodbc==4.0.32
pandas==1.3.4
azure-identity==1.6.1
azure-kusto-data[aio]==3.1.0
azure-storage-file-datalake==12.5.0
packaging==21.2
stringcase==1.2.0
aiohttp==3.8.1

</details>

**Comment by AsafMah** [comment]
Can't recreate - can you give a more full code sample, including what reactor you use to run the async method?

**Comment by AsafMah** [comment]
Closing as there is no response in a few months.

**Comment by jamestrousdale** [comment]
@AsafMah Would you please re-open this?

I think the missing context here is that it does not happen with all authentication methods.

I am currently experiencing exactly this issue with managed identity (workload identity) authentication. Could you see if you could reproduce with that authentication method?

**Comment by AsafMah** [comment]
Thank you for the in depth research!

I'll work on a fix

**Comment by AsafMah** [comment]
Fixed in 4.3.0 - let us know if the issue persists
